packages/fxa-settings/src/components/ConnectAnotherDevicePromo/storeImageLoader.tsx

```diff
@@ -0,0 +1,145 @@
+import koGoogle from '../../assets/images/google_play_store_button/ko.png';
+import ltGoogle from '../../assets/images/google_play_store_button/lt.png';
+import nbNOGoogle from '../../assets/images/google_play_store_button/nb-NO.png';
+import nlGoogle from '../../assets/images/google_play_store_button/nl.png';
+import plGoogle from '../../assets/images/google_play_store_button/pl.png';
+import ptBRGoogle from '../../assets/images/google_play_store_button/pt-BR.png';
+import ptGoogle from '../../assets/images/google_play_store_button/pt.png';
+import ruGoogle from '../../assets/images/google_play_store_button/ru.png';
+import skGoogle from '../../assets/images/google_play_store_button/sk.png';
+import slGoogle from '../../assets/images/google_play_store_button/sl.png';
+import svGoogle from '../../assets/images/google_play_store_button/sv.png';
+import trGoogle from '../../assets/images/google_play_store_button/tr.png';
+import ukGoogle from '../../assets/images/google_play_store_button/uk.png';
+import zhCNGoogle from '../../assets/images/google_play_store_button/zh-CN.png';
+import zhTWGoogle from '../../assets/images/google_play_store_button/zh-TW.png';
+
+export enum StoreType {
+  apple,
+  google,
+}
+
+const storeImages = [
+  {
+    da: daApple,
+    de: deApple,
+    en: enApple,
+    es: esApple,
+    et: etApple,
+    fr: frApple,
+    he: heApple,
+    hu: huApple,
+    id: idApple,
+    it: itApple,
+    ja: jaApple,
+    ko: koApple,
+    lt: ltApple,
+    'nb-NO': nbNOApple,
+    nl: nlApple,
+    pl: plApple,
+    'pt-BR': ptBRApple,
+    pt: ptApple,
+    ru: ruApple,
+    sk: skApple,
+    sl: slApple,
+    'sv-SE': svSEApple,
+    tr: trApple,
+    'zh-CH': zhCNApple,
+    'zh-TW': zhTWApple,
+  },
+  {
+    ca: caGoogle,
+    cs: csGoogle,
+    da: daGoogle,
+    de: deGoogle,
+    en: enGoogle,
+    es: esGoogle,
+    et: etGoogle,
+    fr: frGoogle,
+    hu: huGoogle,
+    id: idGoogle,
+    it: itGoogle,
+    ja: jaGoogle,
+    ko: koGoogle,
+    lt: ltGoogle,
+    'nb-NO': nbNOGoogle,
+    nl: nlGoogle,
+    pl: plGoogle,
+    'pt-BR': ptBRGoogle,
+    pt: ptGoogle,
+    ru: ruGoogle,
+    sk: skGoogle,
+    sl: slGoogle,
+    sv: svGoogle,
+    tr: trGoogle,
+    uk: ukGoogle,
+    'zh-CH': zhCNGoogle,
+    'zh-TW': zhTWGoogle,
+  },
+];
+
+// `PropertyKey` is short for "string | number | symbol"
+// since an object key can be any of those types, our key can too
+// in TS 3.0+, putting just "string" raises an error
+function hasKey<O>(obj: O, key: PropertyKey): key is keyof O {
+  return key in obj;
+}
+
+export function getStoreImageByLanguages(
+  store: StoreType,
+  userLanguages: readonly string[] = ['en']
+) {
+  // Iterate through available languages until logo is found, otherwise default to 'en'
+  for (let i = 0; i < userLanguages.length; i += 1) {
+    // If language string includes region, eg. zh-TW, check if an image is available.
+    const language =
+      userLanguages[i].length === 5 &&
```

Comment:
I think `.includes('-')` and then grabbing the substring before the hyphen will be a little less fragile, we have [tons of locales](https://github.com/mozilla/fxa-content-server-l10n/tree/master/locale) and I'm not sure if all of them will always fit this pattern. I think we may actually be missing some of these locales too which is another reason localizing text on images is not the best but this is currently all we're localizing in the content-server so 😅 

Thanks for the comments in this file btw.

Reply:
Ah good catch. I've updated the logic on this to rather look for the `-`.